src/GovUk.Education.ExploreEducationStatistics.Common/Utils/CsvUtils.cs

```diff
@@ -0,0 +1,329 @@
+    /// of the row in the CSV are provided to the function.
+    /// </summary>
+    /// <remarks>
+    /// This method uses and closes the provided Stream.
+    /// </remarks>
+    /// <param name="streamProvider">The Stream of the CSV.</param>
+    /// <param name="func">The function to execute against each row of the CSV. The function takes the cells and
+    /// the index of the current row, and returns a new value per line of the CSV to generate a new List.
+    /// The index is the zero-based index of the row in the CSV, minus 1 if the CSV header was skipped.
+    /// </param>
+    /// <param name="skipHeaderRow">Choose whether or not to skip a first header row in the executions.</param>
+    public static Task<List<TResult>> Select<TResult>(
+        Func<Task<Stream>> streamProvider,
+        Func<List<string>, int, TResult> func,
+        bool skipHeaderRow = true)
+    {
+        return Select(
+            streamProvider,
+            (cells, index) =>
+            {
+                var result = func.Invoke(cells, index);
+                return Task.FromResult(result);
+            },
+            skipHeaderRow
+        );
+    }
+
+    public static T? BuildType<T>(
+        IReadOnlyList<string> rowValues,
+        List<string> colValues,
+        string column,
+        Func<string, T> func)
+    {
+        var value = Value(rowValues, colValues, column);
+        return value == null ? default : func(value);
+    }
+
+    public static T? BuildType<T>(
+        IReadOnlyList<string> rowValues,
+        List<string> colValues,
+        IEnumerable<string> columns,
+        Func<string[], T> func)
+    {
+        var values = Values(rowValues, colValues, columns);
+        return values.All(value => value == null) ? default : func(values!);
+    }
+
+    public static string?[] Values(IReadOnlyList<string> rowValues, List<string> colValues, IEnumerable<string> columns)
+    {
+        return columns.Select(c => Value(rowValues, colValues, c)).ToArray();
+    }
+
+    public static string? Value(
+        IReadOnlyList<string> rowValues,
+        List<string> colValues,
+        string column,
+        string? defaultValue = null)
+    {
+        if (!colValues.Contains(column))
+        {
+            return defaultValue;
+        }
+
+        var cellValue = rowValues[colValues.FindIndex(h => h.Equals(column))].Trim().NullIfWhiteSpace();
+
+        return cellValue ?? defaultValue;
+    }
+
+    public static GeographicLevel GetGeographicLevel(IReadOnlyList<string> rowValues, List<string> colValues)
+    {
+        var value = Value(rowValues, colValues, "geographic_level");
+
+        try
+        {
+            return (GeographicLevel) GeographicLevelLookup.ConvertFromProvider.Invoke(value)!;
+        }
+        catch (ArgumentOutOfRangeException)
+        {
+            throw new InvalidGeographicLevelException(value!);
+        }
+    }
```

Comment:
I wonder if `GeographicLevelLookup` would be better refactored to being a singleton service in the Importer.

Either that would use `CsvUtils.Value`, or use `CsvUtils.Value` first then `GeographicLevelLookup`.

It would leave CsvUtils a bit purer without referencing a column name here.

Reply:
I've opted to move it out into `GeographicLevelCsvUtils` instead of a singleton service. Going for a singleton service felt a bit overkill.

Of course, if you'd prefer to go with that I can make the change, but would personally prefer to keep the PR as simple as possible.

Reply:
Latest change to add `GeographicLevelCsvUtils` seems like a good way to go, thanks!